src/core/vector/qgsvectorlayer.cpp

```diff
@@ -183,7 +183,12 @@ QgsVectorLayer::QgsVectorLayer( const QString &vectorLayerPath,
   if ( !vectorLayerPath.isEmpty() && !mProviderKey.isEmpty() )
   {
     QgsDataProvider::ProviderOptions providerOptions { options.transformContext };
-    setDataSource( vectorLayerPath, baseName, providerKey, providerOptions, options.loadDefaultStyle );
+    QgsDataProvider::ReadFlags providerFlags = QgsDataProvider::ReadFlags();
+    if ( options.loadDefaultStyle )
+    {
+      providerFlags |= QgsDataProvider::FlagLoadDefaultStyle;
+    }
+    setDataSourcePrivate( vectorLayerPath, baseName, providerKey, providerOptions, providerFlags );
```

Comment:
@m-kuhn my compiler throws a warning here:

![immagine](https://user-images.githubusercontent.com/142164/118009201-2684f200-b34e-11eb-8b82-fff75b4ec081.png)

Reply:
Is the commit https://github.com/qgis/QGIS/pull/41208/commits/dd2edf0bf15f2017f36cc2ac12990261249a0d92 in https://github.com/qgis/QGIS/pull/41208 fixed the warning ? I have replaced `setDataSourcePrivate` call by `setDataSource` ?